qualtran/_infra/gate_with_registers.py

```diff
@@ -371,8 +371,9 @@ def __pow__(self, power: int) -> 'GateWithRegisters':
             return Power(bloq, abs(power))
         raise NotImplementedError(f"{self} does not implemented __pow__ for {power=}.")
 
+    @classmethod
     def _get_ctrl_spec(
-        self,
+        cls,
```

Comment:
where is this called? doesn't the call site need to be updated?

Reply:
It is only used in one location, to convert cirq style controls to `CtrlSpec`:
https://github.com/quantumlib/Qualtran/blob/db8b9938d8210024bb6b92942376363586165efc/qualtran/_infra/gate_with_registers.py#L499-L501

I made this a class method so that I can call it in #1348 (as the code doesn't depend on the instance).

Reply:
can you please update that existing call in `gate_with_registers.py` to call it as a class method, i.e. `GateWithRegisters._get_ctrl_spec` instead of `self.`

Reply:
done